server/api/dataset.py

```diff
@@ -113,26 +120,30 @@ def getDataset(self, dataset, params):
     @access.user
     def ingestDataset(self, params):
         Dataset = self.model('dataset', 'isic_archive')
-        Folder = self.model('folder')
+        File = self.model('file')
         User = self.model('user', 'isic_archive')
 
         if cherrypy.request.headers['Content-Type'].split(';')[0] == \
                 'application/json':
             params = self.getBodyJson()
-        self.requireParams(('uploadFolderId', 'name', 'owner'), params)
+        self.requireParams(('zipFileId', 'name', 'owner'), params)
 
         user = self.getCurrentUser()
         User.requireCreateDataset(user)
 
-        uploadFolderId = params.get('uploadFolderId', None)
-        if not uploadFolderId:
+        zipFileId = params.get('zipFileId')
+        if not zipFileId:
             raise ValidationException(
-                'No files were uploaded.', 'uploadFolderId')
-        uploadFolder = Folder.load(
-            uploadFolderId, user=user, level=AccessType.WRITE, exc=False)
-        if not uploadFolder:
+                'No file was uploaded.', 'zipFileId')
+        zipFile = File.load(
+            zipFileId, user=user, level=AccessType.WRITE, exc=False)
+        if not zipFile:
             raise ValidationException(
-                'Invalid upload folder ID.', 'uploadFolderId')
+                'Invalid upload file ID.', 'zipFileId')
+        if not mimetypes.guess_type(zipFile['name'], strict=False)[0] \
```

Comment:
Let's make the algorithm this:
```
if (zipFile['mimeType'] not in ZIP_FORMATS) and \
   (zipFile['mimeType'] in ['application/octet-stream', None]) and\
   (mimetypes.guess_type(zipFile['name'], strict=False)[0] not in ZIP_FORMATS):
    raise ValidationException(...)
```

This accounts for the fact that files store their own MIME type in the `file['mimeType']` property. However, since that `mimeType` is set by the uploading client, we should fall back to type guessing if `mimeType` is a default or unknown value.